internal/services/oracle/oracle_resource_anchor_resource.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package oracle
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/oracledatabase/2025-09-01/resourceanchors"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ sdk.Resource = ResourceAnchorResource{}
+
+type ResourceAnchorResource struct{}
+
+type ResourceAnchorResourceModel struct {
+	Location            string            `tfschema:"location"`
+	Name                string            `tfschema:"name"`
+	ResourceGroupName   string            `tfschema:"resource_group_name"`
+	LinkedCompartmentID string            `tfschema:"linked_compartment_id"`
+	ProvisioningState   string            `tfschema:"provisioning_state"`
+	Tags                map[string]string `tfschema:"tags"`
+}
+
+func (ResourceAnchorResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.Location(),
+
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"linked_compartment_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
```

Comment:
Please add comments above this field to explain the need for marking it as Optional+Computed.